### ISSUE TITLE
Deep Supervised Hashing with Triplet Labels

### ISSUE THREAD

**Issue by chullhwan-song** [chullhwan-song/Reading-Paper#86]
https://arxiv.org/abs/1612.03900

**Comment by chullhwan-song** [comment]
## Abstract
* Hashing 은 ANN Search에서  popular & powerful 방법
   * for large-scale image retrieval
* 전통적인 방법에서는(DL이 아닌) feature vector를 만들고 난후에 hash화함.
* 이 과정에서 feature vector와 hash code사이에 optimal한 학습으로 인해 생성된것이 아님 
    * end to end가 필요하다는 의미도 포함한듯
* DL 방법은 end to end통해 feature vector와 hash code사이에 학습을 통해 optimal hash code를 발생하는 연구임.
    * 기존 연구에서는 softmax, pairwise label를 통해..
* 이전의 SOTA인 [DPSH](https://github.com/chullhwan-song/Reading-Paper/issues/79)는  pairwise label(loss)를 통해 성취
* 우리의 연구는 triplet loss(label)를 이용함.
* 기본적으로 [DPSH](https://github.com/chullhwan-song/Reading-Paper/issues/79)의 연구에서 영감을 얻어으니 거의 같고, triplet loss만 다른듯

## Contributions
*  triplet label based
* 새로운  hash code 평가
* loss function
* SOTA performance 

## Approach
![image](https://user-images.githubusercontent.com/40360823/52994301-723fde00-345a-11e9-81cc-7decee00e67d.png)
* triplet loss는 많은 paper에서 많이보았기 때문에 넘어감 (밑의 리뷰 참고)
    * Deep metric learning using Triplet network : [[paper]](https://arxiv.org/abs/1412.6622)[[review]](https://github.com/chullhwan-song/Reading-Paper/issues/16)
    * FaceNet: A Unified Embedding for Face Recognition and Clustering : [[paper]](https://arxiv.org/abs/1503.03832)[[review]](https://github.com/chullhwan-song/Reading-Paper/issues/43)
* 다만 여기서는 vector가 아닌 hash code 형태 즉, hamming distance에 의해 기본 distance를 계산.
   * hamming이 아니더래도, hash 를 vector로 생각하여 일반적인 distance(inner product)를 사용해도 무방하다고 생각함.
* Hash Function
   * 정확히 어떤 알고리즘으로 사용했는지는 안나타는듯..
   * refer 정도이니 단순희 sign/or others activation정도를 했을것으로 추정 > 뒤에 sign 함수를 쓰는 내용이.
* Loss Function
![image](https://user-images.githubusercontent.com/40360823/52994831-30179c00-345c-11e9-9e60-08b505fc774f.png)
![image](https://user-images.githubusercontent.com/40360823/52994863-49204d00-345c-11e9-8dba-3be5806ce8b3.png)
![image](https://user-images.githubusercontent.com/40360823/52994929-81c02680-345c-11e9-872b-f4c097d3e043.png)
   * U 는 real vector(fc vector)
## Experiment
![image](https://user-images.githubusercontent.com/40360823/52995003-b7fda600-345c-11e9-9039-060142119ee4.png)
![image](https://user-images.githubusercontent.com/40360823/52994958-98ff1400-345c-11e9-9458-ab4d54778828.png)
![image](https://user-images.githubusercontent.com/40360823/52994982-a6b49980-345c-11e9-8c0f-5e8ab66a7c33.png)
